compiler/semexprs.nim

```diff
@@ -1754,17 +1754,21 @@ proc semReturn(c: PContext, n: PNode): PNode =
   if c.p.owner.kind in {skConverter, skMethod, skProc, skFunc, skMacro} or
       (not c.p.owner.typ.isNil and isClosureIterator(c.p.owner.typ)):
     if n[0].kind != nkEmpty:
-      # transform ``return expr`` to ``result = expr; return``
-      if c.p.resultSym != nil:
+      if n[0].kind == nkAsgn and n[0][0].kind == nkSym and c.p.resultSym == n[0][0].sym:
+        discard "return is already transformed"
+      elif c.p.resultSym != nil:
+        # transform ``return expr`` to ``result = expr; return``
```

Comment:
Maybe the simpler fix is to transform this according to the comment, so `result = expr; return` instead of the current `return result = expr`

Reply:
then how you can avoid double transformation in second sempass?

Reply:
second sempass won't touch the `return` because `n[0].kind == nkEmpty`.

Reply:
I think the tests need to be adapted (macros/tmacrostmt.nim)